reactioncommerce/reaction-component-library
# Update to Babel 7

Impact: **minor**  
Type: **chore**

## Changes
- Updated to Babel 7 packages and config. This involved updating a few of the other packages, too, but I tried to keep it to a minimum.
- I added the Babel optional chaining plugin
- Due to one of the package updates, Jest snapshots no longer output `={undefined}`, so all of the snapshot files have been updated with that change.

## Updating
When pulling these commits, it can be good to completely remove `node_modules` both on the host machine and in the `web` container, both in root and in `/package`, and then run `yarn install` in all of these 4 places. Something like this:

```
rm -rf node_modules
yarn install
cd package
rm -rf node_modules
yarn install
cd ..
docker-compose run --rm web sh
rm -rf node_modules/*
yarn install
cd package
rm -rf node_modules/*
yarn install
exit
dc up
```

## Testing
1. Verify that running (`dc up`) works without errors. Browse the running style guide and verify it looks and works as expected.
1. Verify that `yarn lint` works on the host machine and in the container
1. Verify that `yarn test` works on the host machine and in the container
1. `cd` to `/package` and run `yarn build`. You may see warnings but verify that there are no errors and that the created `dist` folder looks like what we expect to publish to NPM.
1. Verify that you can use optional chaining in a file and it works properly in the style guide and in tests
